### ISSUE TITLE
Naming conventions and abbreviations

### ISSUE THREAD

**Issue by AlicanC** [airbnb/javascript#653]
Naming with abbreviations is tricky. Assume you want a class named after "GNU" and "GPL". How are you supposed to capitalize it?
- GNUGPLClass
- GnuGplClass
- GNUGplClass
- GnuGPLClass

I would go for "GnuGplClass", but W3C uses "GNUGplClass" format. ("XMLHttpRequest")

Airbnb should have a preference here and rules [22.2](https://github.com/airbnb/javascript#22.2) and [22.3](https://github.com/airbnb/javascript#22.3) should be extended accordingly.

**Comment by ljharb** [comment]
imo acronyms and initialisms should always be all capitalized, or all lowercased - there's no benefit in making things less readable to satisfy an outdated camelCasing algorithm.

It definitely gets tricky with doubled initialisms, like "XML" and "HTTP", or "GNU" and "GPL". However, typically there, the solution is to use a better name. Since I don't know the use case of your "GNU + GPL" thing, I'll use XMLHttpRequest - clearly "httpRequest" would be both a more accurate and less tricky name for a thing that is rarely used with XML :-)

In other words, when a desired name would force one to do something awkward with capitalization, my advice is to rethink that name.

**Comment by BMorearty** [comment]
I like the suggestion to use initial caps whenever an acronym is used--e.g., XmlHttpRequest or GnuGplClass. It's what I've used for years. It works out nicely.

**Comment by AlicanC** [comment]
> I like the suggestion to use initial caps whenever an acronym is used--e.g., XmlHttpRequest or GnuGplClass. It's what I've used for years. It works out nicely.

That's also what I've been using, but I started to question the style after I realized W3C does it differently.

**Comment by ljharb** [comment]
The name of `XMLHttpRequest` is a legacy from its original creation in IE, and isn't reflective of the W3C's naming conventions.

**Comment by AlicanC** [comment]
What does W3C use now?

**Comment by ljharb** [comment]
http://www.w3.org/2005/rules/wg/wiki/Arch/Naming_Conventions.html looks to be their conventions. However, I wouldn't look to the W3C for best practices regarding naming conventions :-)

**Comment by incleaf** [comment]
Then should I use `XMLHTTPRequest` since there is no rule for doubled initialisms?

**Comment by acjh** [comment]
there's this:

> - [23.9](https://github.com/airbnb/javascript#naming--Acronyms-and-Initialisms) Acronyms and initialisms should always be all capitalized, or all lowercased.

**Comment by ljharb** [comment]
This seems answered.
